packages/npm-resolver/src/pickPackage.ts

```diff
@@ -202,3 +204,9 @@ function toUri (pkgName: string, registry: string) {
 
   return url.resolve(registry, encodedName)
 }
+
+function validatePackageName (pkgName: string) {
+  if (pkgName.includes('/') && pkgName[0] !== '@') {
+    throw new Error(`Package name ${pkgName} is invalid, it should have a @scope`)
```

Comment:
This should be a PnpmError with some error code
```suggestion
    throw new PnpmError('INVALID_PACKAGE_NAME', `Package name ${pkgName} is invalid, it should have a @scope`)
```